src/dagmc/DagMC.hpp

```diff
@@ -0,0 +1,460 @@
+#ifndef MOABMC_HPP
+#define MOABMC_HPP
+
+#include "MBTagConventions.hpp"
+#include "moab/CartVect.hpp"
+#include "moab/Range.hpp"
+#include "moab/Core.hpp"
+#include "moab/GeomUtil.hpp"
+#include "moab/FileOptions.hpp"
+#include "moab/Interface.hpp"
+#include "moab/GeomTopoTool.hpp"
+#include "moab/GeomQueryTool.hpp"
+#include "DagMCVersion.hpp"
+
+#include <vector>
+#include <map>
+#include <string>
+#include <assert.h>
+
+class RefEntity;
+
+struct DagmcVolData {
+  int mat_id;
+  double density, importance;
+  std::string comp_name;
+};
+
+
+namespace moab {
+
+class CartVect;
+
+/**\brief
+ *
+ * In sectction 1, the public interface you will find all the functions needed
+ * for problem setup. For the typical MC code, the order of function calls
+ * required to fully populate DAGMC ready to run are
+ *
+ *    1) DAG->load_file();
+ *    2) DAG->init_OBBTree();
+ *
+ * Modifications were made to init_OBBTree which allows the functions of
+ * init_OBBTree to be called without having used init_OBBTree. For example
+ * if you would like access to be able to call DAG->point_in_volume() but without
+ * having an implicit compliment you need only call
+ *
+ *   1) DAG->load_file();
+ *   2) DAG->setup_obb();
+ *
+ * Similarly, if you need access to problem indices only then, one may call
+ * load_file followed by setup_indices.
+ *
+ *   1) DAG->load_file();
+ *   2) DAG->setup_indices();
+ */
+
+class DagMC {
+ public:
+  // Constructor
+  DagMC(Interface* mb_impl = NULL, double overlap_tolerance = 0., double numerical_precision = .001);
```

Comment:
Optional: Similar to how `const char*` is used on line 89, it may be valuable to use `const` when declaring other method parameters, especially pointers.  This makes it easier for the compile to find accidental re-assignment of parameter values.